daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ZonedTextParsers.scala

```diff
@@ -17,14 +17,421 @@
+  val min = Int.MinValue.toLong
+  val max = Int.MaxValue.toLong
+}
+
+case class ConvertZonedShortParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[Short]() {
+
+  override def getNum(num: Number) = num.shortValue
+  override val xsdType = "short"
+  override val prettyType = "Short Integer"
+  val min = Short.MinValue.toLong
+  val max = Short.MaxValue.toLong
+}
+
+case class ConvertZonedByteParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[Byte]() {
+
+  override def getNum(num: Number) = num.byteValue
+  override val xsdType = "byte"
+  override val prettyType = "Byte"
+  val min = Byte.MinValue.toLong
+  val max = Byte.MaxValue.toLong
+}
+
+case class ConvertZonedUnsignedLongParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[JBigInt]() {
+
+  override def getNum(num: Number) = new JBigInt(num.toString)
+  override val xsdType = "unsignedLong"
+  override val prettyType = "Unsigned Long"
+  override def isInvalidRange(jn: java.lang.Number) = {
+    jn match {
+      case n: JBigInt => {
+        n.compareTo(JBigInt.ZERO) < 0 || n.compareTo(JBigInt.ONE.shiftLeft(64)) >= 0
+      }
+      case _ => {
+        val n = jn.longValue()
+        n < 0 // note: the other side of the check is inherently ok since a Long must be smaller than an unsignedLong.
+      }
+    }
+  }
+  val min = 0.toLong
+  val max = -1.toLong // unused.
+}
+
+case class ConvertZonedUnsignedIntParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[Long]() {
+
+  override def getNum(num: Number) = num.longValue
+  override val xsdType = "unsignedInt"
+  override val prettyType = "Unsigned Integer"
+  val min = 0L
+  val max = (1L << 32) - 1L
+}
+
+case class ConvertZonedUnsignedShortParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[Int]() {
+
+  override def getNum(num: Number) = num.intValue
+  override val xsdType = "unsignedShort"
+  override val prettyType = "Unsigned Short"
+  val min = 0L
+  val max = (1L << 16) - 1L
+}
+
+case class ConvertZonedUnsignedByteParserUnparserHelper[S]()
+  extends ConvertZonedIntegerNumberParserUnparserHelper[Short]() {
+
+  override def getNum(num: Number) = num.shortValue
+  override val xsdType = "unsignedByte"
+  override val prettyType = "Unsigned Byte"
+  val min = 0L
+  val max = (1L << 8) - 1L
+}
+
+case class ConvertZonedDecimalParserUnparserHelper[S]()
+  extends ConvertZonedFloatingPointNumberParserUnparserHelper[JBigDecimal]() {
+
+  override def getNum(num: Number) = new JBigDecimal(num.toString)
+  override val xsdType = "decimal"
+  override val prettyType = "Unlimited Size Decimal"
+  override def isInvalidRange(n: java.lang.Number): Boolean = false
+
+  override def getStringFormat(n: JBigDecimal): String = {
+    n.toPlainString()
+  }
+}
+
+abstract class ZonedFormatFactoryBase[S](parserHelper: ConvertZonedNumberParserUnparserHelperBase[S]) extends Serializable {
```

Comment:
This actually looks really similar to the standard text factory base. Can the zoned factory just extend that and pass in Maybe.Nope for properties that don't apply to zoned. I think the only difference is that the "+" sign needs to be stripped off of pattern?